### ISSUE TITLE
'data.relationships.data' has all attributes of relationship by default when it shouldn't

### ISSUE THREAD

**Issue by kurko** [rails-api/active_model_serializers#2027]
#### Expected behavior vs actual behavior

I created 2 models and added `has_one` to the serializer, by default, it returns the relationship object:

```json
{
  "data": {
    "id": "1", 
    "type": "posts", 
    "attributes": { "title": "my title" }, 
    "relationships": {
      "comment": { "data": { "id": "2", "name": "John", "text": "whatever" } }
    }
  }
}
```

However, I was expecting it to be

```json
{
  "data": {
    "id": "1", 
    "type": "posts", 
    "attributes": {
       "title": "my title"
    }, 
    "relationships": {
      "comment": { "data": { "id": "2", "type": "comment" } }
    }
  }
}
```

JSONAPI specifies that

> The value of the relationships key MUST be an object (a “relationships object”).

See what this means here: http://jsonapi.org/format/#document-resource-object-linkage

Basically, if I passed `included`, then I'd expect to see the actual objects. Otherwise it only makes sense to return the Object Linkage (which is what JSONAPI specifies anyway).

#### Steps to reproduce

Just create 2 serializers and add a `has_one` to one of them.

#### Environment

ActiveModelSerializers Version: 0.10.4

Integrated application and version *(e.g., Rails, Grape, etc)*: Rails

**Comment by kurko** [comment]
Ok, found the bug. The behavior above happens when there are 2 models, 1 serializer. When I add a serializer for the second model, it shows and object linkage. Buggy nonetheless.

**Comment by bf4** [comment]
@kurko I have an idea why that's happening.. the reflection -> association -> relationship -> resourceidentifier has some dependencies on the serializer.. for a 'virtual value' it does `object.as_json || object`. The JSONAPI adapter shouldn't allow raw objects as associations unless it's going to try to figure it out like `{ type: object.class.name, object.id }`

**Comment by bf4** [comment]
@kurko what do you think the expected behavior to be? JSON API associations require a serializer?

**Comment by kurko** [comment]
I don't think we should require serializers. Instead, we should just _always_ whitelist `type`, `id`, `links` etc (the fields that are acceptable according to the standard). That would wipe out keys that shouldn't be there.

**Comment by bf4** [comment]
@kurko 

> type, id, links

where would 'type', 'id',  come from, though? In #1857 (which is being superseded by #2026) I did

```ruby
 if belongs_to_with_foreign_key?(association)
   id = parent_serializer.object.send(association.reflection.foreign_key)
   type = association.reflection.type.to_s
```

I guess, if you're ok instantiating the association object, you can do something like `id = association.object.id`
